tests/test_api.py

```diff
@@ -1,24 +1,27 @@
+import typing
+
 import pytest
+from uvicorn.main import Server
 
 import httpx
 
 
-def test_get(server):
+def test_get(server: Server) -> None:
     response = httpx.get(server.url)
     assert response.status_code == 200
     assert response.reason_phrase == "OK"
     assert response.text == "Hello, world!"
     assert response.http_version == "HTTP/1.1"
 
 
-def test_post(server):
+def test_post(server: Server) -> None:
     response = httpx.post(server.url, data=b"Hello, world!")
     assert response.status_code == 200
     assert response.reason_phrase == "OK"
 
 
-def test_post_byte_iterator(server):
-    def data():
+def test_post_byte_iterator(server: Server) -> None:
+    def data() -> typing.Generator[bytes, None, None]:
```

Comment:
Can we prefer this more relaxed style, in line with our other typing usage in this repo?

```suggestion
    def data() -> typing.Iterator[bytes]:
```